pulp_ansible/app/migrations/0047_add_repo_cv_index.py

```diff
@@ -0,0 +1,155 @@
+from django.contrib.postgres import search as psql_search
+from django.db import migrations
+
+
+#  A view that does a view things:
+#   * subquery to make a count of signatures attached to the CV and the repo
+#   * subquery to make a count of deprecations related to the CV
+#   * The app code has an unmanaged model that makes this view look like any
+#     other model and is compatible with filtersets and serializers.
+#
+# Why?
+#   * The x-repo search requirements want the same CV listed N times per
+#     repositories containing the CV.
+#   * Many models and objects and concepts desired for cross repo search
+#     filtering are not directly related to repositories.
+#   * Using the queryset language to build the counts for signatures and
+#     deprecations were drifting from wizardry into impossibility.
+#   * Creating a real index model requires too many hooks in various
+#     obscure workflows for CRUD and new devs will always make mistakes.
+
+CREATE_REPOSITORY_COLLECTIONVERSION_VIEW = '''
+DROP VIEW IF EXISTS repository_collectionsignature;
+CREATE OR REPLACE VIEW repository_collectionsignature AS
+
+SELECT
+
+    cr.pulp_id as repository_id,
+    cr.name,
+    acvs.content_ptr_id,
+    acvs.signed_collection_id as collectionversion_id,
+
+    MAX(crv1.number) as sig_vadded,
+    coalesce(MAX(crv2.number), -1) as sig_vremoved,
+
+    True as is_signed
+
+from
+    core_repositorycontent crc
+left join
+    ansible_collectionversionsignature acvs on acvs.content_ptr_id=crc.content_id
+left join
+    core_repository cr on cr.pulp_id=crc.repository_id
+left join
+    core_repositoryversion crv1 on crv1.pulp_id=crc.version_added_id
+left join
+    core_repositoryversion crv2 on crv2.pulp_id=crc.version_removed_id
+left join
+    ansible_collectionversion acv on acv.content_ptr_id=acvs.content_ptr_id
+WHERE
+    acvs.content_ptr_id is not null
+GROUP BY
+    cr.pulp_id, acvs.content_ptr_id
+HAVING
+    MAX(crv1.number) > coalesce(MAX(crv2.number), -1)
+;
```

Comment:
This is neat logic, should you replace this with the defined names above `sig_vadded` and `sig_vremoved`?

Reply:
I believe it is allowed in regular sql, but for whatever reason django doesn't like it ....

```
django.db.utils.ProgrammingError: column "sig_vadded" does not exist
```